app.py

```diff
@@ -88,7 +89,17 @@ def route(album, artist):
 
 @app.route("/artists/<artist_name>", methods=("GET", "POST"))
 def get_single_artist(artist_name):
-    return render_template("single_artist.html")
+    artist_albums = []
```

Comment:
Everything looks good, bu I am going to ask for a small change here. 

Can this code that organizes the data returned from the query be moved to the Artist table? For example, AlbumTable has a function called _album_data: https://github.com/ecedmondson/pitchfork_copycat/blob/4c59691a37051741b6c780dfcbec660ebf372752/database/models/tables.py#L41

_album_data has a similar purpose to your code here: organize data returned from the query so that the data is meaningful to us. However, that type of operation more properly "belongs" to the AlbumTable than to the app. That is, the code for the route shouldn't have to "care" about how to format and organize the data, just how to pass that data to the right spot. 

I think making a new functon in ArtistTable like _format_data_for_single_artist_query (or some other function name; i don't care what the name is)  and moving this code there will probably be for the best. 

The code is absolutely fine how it is, btw. The only thing that caught my attention is that you're not putting the albums on the query object (or, in your case, query=single_artist object). I think you could probably do it in dictionary format for a little more clarity. However, what you have here works and is readable so I wouldn't consider that a must_change. You can either copy and paste it directly or try and use a dictionary. Whichever works.